mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3746,6 +3748,30 @@ int main(int argc, char **argv) {
   llvm::cl::ParseCommandLineOptions(argc, argv,
                                     "MLIR Rock Dialect host generation\n");
 
+  if (!arch.getValue().empty()) {
+    bool archPrefersOCP = (archChip().substr(0, 5) == "gfx12");
+    std::map<F8TypesChoice, std::string> f8e4m3TypeNames{
```

Comment:
LLVM's types? `DenseMap<F8TypesChoice, StringLiteral>` or something like that?

Reply:
DenseMap yes, StringLiteral no -- there's no implicit constructor for StringLiteral from "", and also the map is used to set some cl::opts that are themselves std::string.

Reply:
Fair enough, so we're probably good as is